File: 2657. Find the Prefix Common Array of Two Arrays/Solution.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size(), curr = 0;
        vector<int> res(n), counter(51);
        for (int i = 0; i < n; i++) {
            if (counter[A[i]] < 0) curr++;
            counter[A[i]]++;
            if (counter[B[i]] > 0) curr++;
            counter[B[i]]--;
            res[i] = curr;
        }
        return res;
    }
};

File: 2657. Find the Prefix Common Array of Two Arrays/Solution.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        res, counter, curr = [], defaultdict(int), 0
        for x, y in zip(A, B):
            if counter[x] < 0: curr += 1
            counter[x] += 1
            if counter[y] > 0: curr += 1
            counter[y] -= 1
            res.append(curr)
        return res